### PR TITLE
AE-1279: Checkbox Component

### PR DESCRIPTION
**Pull request by williamjs95**
## Summary by CodeRabbit

- **New Features**
  - Introduced a customizable CheckBox component with support for multiple sizes, visual states, themes, error and helper text, and accessibility features.
  - Added the CheckBox component to the public component library for use in applications.

- **Documentation**
  - Provided interactive stories demonstrating the CheckBox component’s features, states, and theme variations.

- **Tests**
  - Added a comprehensive test suite covering rendering, interaction, accessibility, and edge cases for the CheckBox component.

### CODE REVIEW

File: src/components/CheckBox/CheckBox.test.tsx [hunk 1]
```diff
@@ -0,0 +1,786 @@
+import React from 'react';
+import { render, screen, fireEvent } from '@testing-library/react';
```
[the remark this answers is not in the document]
Comment (souzajr) [reply src/components/CheckBox/CheckBox.test.tsx:1]: Feito

**Review by souzajr — APPROVED** [review]
